### PR TITLE
Static typing: Historical funding and fixing issues with isolated market orders

### PR DESCRIPTION
**Pull request by ruixhuang**
Tested on Android and iOS.

### CODE REVIEW

File: src/commonMain/kotlin/exchange.dydx.abacus/calculator/ReceiptCalculator.kt [hunk 1]
```diff
@@ -68,7 +68,7 @@ internal class ReceiptCalculator {
                             listOf(
                                 ReceiptLine.Equity,
                                 ReceiptLine.MarginUsage,
-                                ReceiptLine.Fee,
+                                ReceiptLine.TransferFee,
```
Comment (ruixhuang) [inline src/commonMain/kotlin/exchange.dydx.abacus/calculator/ReceiptCalculator.kt:71]: For mobile only.  FE web doesn't use this.

File: src/commonMain/kotlin/exchange.dydx.abacus/calculator/V2/SubaccountTransformerV2.kt [hunk 2]
```diff
@@ -193,11 +193,12 @@ internal class SubaccountTransformerV2(
         hasTransfer: Boolean = false,
     ) {
         val deltaMarketId = delta?.marketId
-        val positions = subaccount.openPositions
+        val positions = subaccount.openPositions ?: mapOf()
```
Comment (ruixhuang) [inline src/commonMain/kotlin/exchange.dydx.abacus/calculator/V2/SubaccountTransformerV2.kt:196]: Fixing issue when create a new subaccount for isolated margin.

File: src/commonMain/kotlin/exchange.dydx.abacus/calculator/V2/TransferInputCalculatorV2.kt [hunk 3]
```diff
@@ -126,9 +126,9 @@ internal class TransferInputCalculatorV2(
 
                 return TransferInputSummary(
                     filled = true,
-                    gasFee = transfer.fee,
+                    gasFee = null,
                     usdcSize = usdcSize,
-                    fee = null,
+                    fee = transfer.fee,
```
Comment (ruixhuang) [inline src/commonMain/kotlin/exchange.dydx.abacus/calculator/V2/TransferInputCalculatorV2.kt:131]: For mobile only.  FE web doesn't use this.

File: src/commonMain/kotlin/exchange.dydx.abacus/output/input/ReceiptLine.kt [hunk 4]
```diff
@@ -27,7 +27,8 @@ enum class ReceiptLine(val rawValue: String) {
     CrossMarginUsage("CROSS_MARGIN_USAGE"),
     PositionMargin("POSITION_MARGIN"),
     PositionLeverage("POSITION_LEVERAGE"),
-    LiquidationPrice("LIQUIDATION_PRICE");
+    LiquidationPrice("LIQUIDATION_PRICE"),
+    TransferFee("TRANSFER_FEE");
```
Comment (ruixhuang) [inline src/commonMain/kotlin/exchange.dydx.abacus/output/input/ReceiptLine.kt:31]: For mobile only.  FE web doesn't use this.

File: src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/OnboardingSupervisor.kt [hunk 5]
```diff
@@ -865,13 +865,22 @@ internal class OnboardingSupervisor(
...
 
     private fun receiveTransferGas(gas: BigDecimal?) {
-        val input = stateMachine.input
-        val oldFee = helper.parser.asDecimal(helper.parser.value(input, "transfer.fee"))
-        if (oldFee != gas) {
-            val oldState = stateMachine.state
-            val modified = input?.mutable() ?: iMapOf<String, Any>().mutable()
-            modified.safeSet("transfer.fee", gas)
-            update(StateChanges(iListOf(Changes.input)), oldState)
+        if (stateMachine.staticTyping) {
```
Comment (ruixhuang) [inline src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/OnboardingSupervisor.kt:868]: Adding processing for transfer gas

**Review by tyleroooo — DISMISSED** [review]
I really don't know what's going on here. My approval means very little in this case.

**Review by tyleroooo — APPROVED** [review]
continuing to yolo stamp

**Pull request merged**